### PR TITLE
ENG-8026 updated voltdb mask to use sha256

### PR DESCRIPTION
**Pull request by dweiss88**

### CODE REVIEW

File: src/frontend/org/voltdb/client/ClientAuthHashScheme.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
  * @author akhanzode
  */
 public enum ClientAuthHashScheme {
-    HASH_SHA1(0), HASH_SHA256(1);
+    HASH_SHA1(0), HASH_SHA256(1), HASH_SHABOTH(2);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/client/ClientAuthHashScheme.java]: Can you not add this in enum as its not really an enum.

File: src/frontend/org/voltdb/client/ClientAuthHashScheme.java [hunk 2]
```diff
@@ -63,6 +63,7 @@ public final static int getHexencodedDigestLength(ClientAuthHashScheme scheme) {
         switch (scheme) {
             case HASH_SHA1 : return 40;
             case HASH_SHA256 : return 64;
+            case HASH_SHABOTH : return 104;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/client/ClientAuthHashScheme.java]: Remove this from here.

File: src/frontend/org/voltdb/compiler/VoltProjectBuilder.java [hunk 3]
```diff
@@ -139,6 +139,14 @@ public boolean equals(final Object o) {
         public final String name;
         public String password;
         private final String roles[];
+        public boolean plaintext = true;
```
Comment (akhanzode) [inline src/frontend/org/voltdb/compiler/VoltProjectBuilder.java]: Do u need to modify project builder? You can leave it as is.

File: src/frontend/org/voltdb/utils/CatalogUtil.java [hunk 4]
```diff
@@ -1486,17 +1486,22 @@ private static void setUsersInfo(Catalog catalog, UsersType users) {
             if (user.isPlaintext()) {
                 sha1hex = extractPassword(user.getPassword(), ClientAuthHashScheme.HASH_SHA1);
                 sha256hex = extractPassword(user.getPassword(), ClientAuthHashScheme.HASH_SHA256);
+            } else if (user.getPassword().length() == ClientAuthHashScheme.getHexencodedDigestLength(ClientAuthHashScheme.HASH_SHABOTH)) {
```
Comment (akhanzode) [inline src/frontend/org/voltdb/utils/CatalogUtil.java]: Check a hardcoded length. if length does not match throw error and don't start the server.

File: src/frontend/org/voltdb/utils/CatalogUtil.java [hunk 5]
```diff
@@ -1486,17 +1486,22 @@ private static void setUsersInfo(Catalog catalog, UsersType users) {
...
+                int sha1len = ClientAuthHashScheme.getHexencodedDigestLength(ClientAuthHashScheme.HASH_SHA1);
+                sha1hex = sha1hex.substring(0, sha1len);
+                sha256hex = sha256hex.substring(sha1len);
             }
             org.voltdb.catalog.User catUser = db.getUsers().add(user.getName());
 
+            String saltGen = BCrypt.gensalt(BCrypt.GENSALT_DEFAULT_LOG2_ROUNDS,sr);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/utils/CatalogUtil.java]: Add a comment here.

**Comment by dweiss88** [comment]
squash merged

**Pull request closed without merging**